scripts/spack/configs/toss_4_x86_64_ib_cray/spack.yaml

```diff
@@ -32,6 +32,81 @@ spack:
         fc: /opt/rocm-5.2.3/llvm/bin/amdflang
       spec: clang@14.0.0
       target: x86_64
+  - compiler:
+      environment: {}
+      extra_rpaths: []
+      flags:
+        fflags: -Mfreeform
+      modules: []
+      operating_system: rhel8
+      paths:
+        cc: /opt/rocm-5.4.3/llvm/bin/amdclang
+        cxx: /opt/rocm-5.4.3/llvm/bin/amdclang++
+        f77: /opt/rocm-5.4.3/llvm/bin/amdflang
+        fc: /opt/rocm-5.4.3/llvm/bin/amdflang
+      spec: clang@15.0.0
+      target: x86_64
+  - compiler:
+      environment: {}
+      extra_rpaths: []
+      flags:
+        fflags: -Mfreeform
+      modules: []
+      operating_system: rhel8
+      paths:
+        cc: /opt/rocm-5.5.1/llvm/bin/amdclang
+        cxx: /opt/rocm-5.5.1/llvm/bin/amdclang++
+        f77: /opt/rocm-5.5.1/llvm/bin/amdflang
+        fc: /opt/rocm-5.5.1/llvm/bin/amdflang
+      spec: clang@16.0.0
+      target: x86_64
+  - compiler:
+      environment: {}
+      extra_rpaths: []
+      # Flag for lowercase Fortran module names
+      flags:
+        fflags: -ef
+      modules: []
+      operating_system: rhel8
+      paths:
+        cc: /usr/tce/packages/cce-tce/cce-15.0.1/bin/craycc
+        cxx: /usr/tce/packages/cce-tce/cce-15.0.1/bin/crayCC
+        f77: /usr/tce/packages/cce-tce/cce-15.0.1/bin/crayftn
+        fc: /usr/tce/packages/cce-tce/cce-15.0.1/bin/crayftn
+      spec: cce@15.0.1
+      target: x86_64
+
+  # One compiler to test
+  - compiler:
+      environment: {}
+      extra_rpaths: []
+      flags:
+        fflags: -Mfreeform
+      modules: []
+      operating_system: rhel8
+      paths:
+        cc: /usr/tce/packages/rocm/rocm-5.6.0beta1/bin/amdclang
+        cxx: /usr/tce/packages/rocm/rocm-5.6.0beta1/bin/amdclang++
+        f77: /usr/tce/packages/rocm/rocm-5.6.0beta1/bin/amdflang
+        fc: /usr/tce/packages/rocm/rocm-5.6.0beta1/bin/amdflang
+      spec: clang@16.0.1
+      target: x86_64
+
+  # Another one (with wrapper)
+  - compiler:
+      environment: {}
+      extra_rpaths: []
+      flags:
+        fflags: -Mfreeform
+      modules: []
+      operating_system: rhel8
+      paths:
+        cc: /usr/tce/packages/rocmcc/rocmcc-5.6.0beta1-magic/bin/amdclang
+        cxx: /usr/tce/packages/rocmcc/rocmcc-5.6.0beta1-magic/bin/amdclang++
+        f77: /usr/tce/packages/rocmcc/rocmcc-5.6.0beta1-magic/bin/amdflang
+        fc: /usr/tce/packages/rocmcc/rocmcc-5.6.0beta1-magic/bin/amdflang
+      spec: clang@16.0.2
+      target: x86_64
```

Comment:
What's the difference between these two compilers?

Reply:
> What's the difference between these two compilers?

From what I remember, the "magic" version contains LC-specific paths added and the other does not. These compilers and the associated host-configs are not fully working yet, and have __host__ __device__ related error:

```
In file included from axom/src/thirdparty/tests/cli11_smoke.cpp:12:
axom/src/thirdparty/axom/CLI11.hpp:4749:47: error: reference to __device__ function 'operator delete' in __host__ function
    std::shared_ptr<FormatterBase> formatter_{new Formatter()};
                                              ^
axom/src/thirdparty/tests/cli11_smoke.cpp:33:18: note: called by 'main'
  axom::CLI::App app {sstr.str()};
                 ^
/collab/usr/global/tools/tce4/packages/rocm/rocm-5.6.0beta1/llvm/lib/clang/16.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
In file included from axom/src/thirdparty/tests/cli11_smoke.cpp:12:
axom/src/thirdparty/axom/CLI11.hpp:4856:47: error: reference to __device__ function 'operator delete' in __host__ function
    std::shared_ptr<Config> config_formatter_{new ConfigINI()};
                                              ^
/collab/usr/global/tools/tce4/packages/rocm/rocm-5.6.0beta1/llvm/lib/clang/16.0.0/include/cuda_wrappers/new:63:24: note: 'operator delete' declared here
__device__ inline void operator delete(void* ptr) CUDA_NOEXCEPT {
                       ^
2 errors generated when compiling for host.
make[2]: *** [thirdparty/tests/CMakeFiles/cli11_smoke_test.dir/build.make:76: thirdparty/tests/CMakeFiles/cli11_smoke_test.dir/cli11_smoke.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3001: thirdparty/tests/CMakeFiles/cli11_smoke_test.dir/all] Error 2
```

Reply:
Tom Stitt ran into a similar issue with this beta compiler. I recommend that we hold off on this until the official 5.6 ROCm release is available.

Reply:
@tomstitt asked us to look into why axom wasn't building with rocm@5.6.0beta1 -- see https://github.com/LLNL/axom/issues/1131 -- but only really cares about the official 5.6 release.

@rhornung67 -- do you have reason to believe that this won't be a problem with the official release?

Reply:
Thanks @bmhan12 -- for some reason I focused on the `rocm` vs `rocmcc` part of the path and missed the "magic" part. 

Should we label the compilers with "magic" (in the spack.yaml file) to better differentiate the resultant host-configs? As is, it's not too clear why people would choose the `clang@16.0.1` vs. `clang@16.0.2` compilers based only on the host-config name.

Reply:
@kennyweiss i do not know if the issue with the beta compiler is known to the AMD folks so that it can be resolved for the official release. I will ping them in our RAJA-AMD group chat.

Reply:
When rocm@5.6.0 is released, these compiler configs will be going away in favor of the new release. 

> Should we label the compilers with "magic" (in the spack.yaml file) to better differentiate the resultant host-configs?

Agreed, magic compilers should have like a `_magic` suffix.

I have the magic and non-magic version here for now just for testing, but I generally prefer the non-magic compilers because the magic relative-path additions feel like they have potential to not play nicely with spack. 
I expect to have just one `clang@16.0.0` compiler host-config with rocm@5.6.0.

Though, separately, there is a working host-config with clang@16.0.0 and rocm@5.5.
Would it make sense to have both a host-config with clang@16.0.0, rocm@5.5 and a host-config with clang@16.0.0, rocm@5.6?

Reply:
IMO, I think we should be testing the latest rocm release when it is available. We can poll our users to see which prior versions they are using and test what they are using.

Reply:
The 5.6.0 release will be installed on rzvernal and tioga tomorrow. We've been asked to report whether the problem persists when we try that.